### DISCUSSION TITLE
Time difference between motion stream and camera stream

### DISCUSSION THREAD

**Discussion by JM-HH** [Motion-Project/motion#1655] [Q&A] [answered]
Hello,
I have two cameras (Bosch Dinion Starlight 6000) which are working fine until I upgraded my PC and motion.
One of the cameras is working as aspected with motion 4.5.1, the other camera shows a increasing time difference between the stream provided by motion and the stream comes directly from the camera. After 12h the time difference is about 5 min and the recorded videos are very small with a lot of artefacts. The log is filled with the following messeages

[2:nc2:Hinten] [INF] [ENC] [Mar 20 19:27:21] ffmpeg_avcodec_log: CSeq 10656 expected, 10643 received.
[3:nc3:Hinten] [INF] [ENC] [Mar 20 19:27:30] ffmpeg_avcodec_log: CSeq 10655 expected, 10645 received.

I used exactly the same configuration for both cameras and motion. 
With motion 4.3 both cameras are working. 
The timestamp in the picture is generated by the camera, the stream in VLC shows the correct time and the stream from motion is 2 min. behind.
![21-03-23_12-42_677](https://user-images.githubusercontent.com/82896633/226596466-48519a26-ff0c-436b-962f-3fa5f7814930.png)

Any ideas
JM

**Comment by tosiara** [comment]
I have something similar with one of my RTSP cameras. In my case the video in motion gets behind for up to 45 seconds. But I haven't found a solution yet

Can you try to add this param to your config?

`netcam_params fflags=nobuffer`

**Comment by Mr-Dave** [comment, marked as answer]
Also try increasing/specifying the `capture_rate` for the camera.

**Reply by tosiara** [reply]
Is it by default FPS + 1?
```
[1:ml1:a] [INF] [NET] netcam_rtsp_read_image: norm: capture_rate not specified in netcam_params. Using 11
[2:nc2:a] [INF] [NET] netcam_rtsp_connect: norm: Netcam capture_rate is 11.
[2:nc2:a] [INF] [NET] netcam_rtsp_connect: norm: Camera source is 10 FPS
```

**Reply by JM-HH** [reply]
I adjusted the capture_rate to the camera framerate +1, this fixed the problem. If the camera framerate is equal to the capture_rate there is still an increasing time difference.

**Comment by Mr-Dave** [comment]
Yes.  But sometimes the camera does not provide a FPS and Motion falls back to the detection frame rate.  When the times fall behind, the pace of the capture and decoding needs to speed up.

**Comment by JM-HH** [comment]
Thank you for our input. 
I added the fflags=nobuffer and increased the capture_rate. After restart of motion the time difference is 1 sec. so I will wait til tommorow if the difference is increasing.
